bus/bus.go

```diff
@@ -1771,19 +1777,26 @@ func (b *bus) paramsHandlerUploadGET(jc jape.Context) {
 	})
 }
 
-func (b *bus) consensusState() api.ConsensusState {
-	cs := b.cm.TipState()
+func (b *bus) consensusState(ctx context.Context) (api.ConsensusState, error) {
+	tip := b.cm.TipState()
+
+	index, err := b.cs.ChainIndex(ctx)
+	if err != nil {
+		return api.ConsensusState{}, err
+	}
 
 	var synced bool
-	if block, ok := b.cm.Block(cs.Index.ID); ok && time.Since(block.Timestamp) < 2*cs.BlockInterval() {
+	block, found := b.cm.Block(index.ID)
+	if found && time.Since(block.Timestamp) < 2*tip.BlockInterval() {
 		synced = true
 	}
 
 	return api.ConsensusState{
-		BlockHeight:   cs.Index.Height,
-		LastBlockTime: api.TimeRFC3339(cs.PrevTimestamps[0]),
-		Synced:        synced,
-	}
+		BlockHeight:      tip.Index.Height,
+		LastBlockTime:    api.TimeRFC3339(tip.PrevTimestamps[0]),
+		Synced:           synced,
+		SubscriberHeight: index.Height,
```

Comment:
Not sure if having 2 heights isn't a bit too confusing. At the end of the day the height that matters is probably the height we have synced to so I'd consider getting rid of `SubscriberHeight` and return `BlockHeight: index.Height` while also making `synced` dependant on `index` rather than `tip`.